File: 10-greedy/08_boats.c
#include "../test_runner.h"

int num_rescue_boats(int* people, int n, int limit) {
    return -1;
}

int main() {
    TEST_INT("pair",   num_rescue_boats(ARR(1,2), 2, 3),    1);
    TEST_INT("mixed",  num_rescue_boats(ARR(3,2,2,1), 4, 3),3);
    TEST_INT("heavy",  num_rescue_boats(ARR(3,5,3,4), 4, 5),4);
    TEST_SUMMARY();
}

File: 10-greedy/08_boats.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Boats to Save People (LC #881) -- Medium
# Min boats, each holds 2 people with weight limit.
#
#   Key: Sort. Pair lightest with heaviest. If they fit, pair them. Else heaviest goes alone.

def num_rescue_boats(people, limit):
    pass


run_tests(num_rescue_boats, [
    (([1,2], 3),      1),
    (([3,2,2,1], 3),  3),
    (([3,5,3,4], 5),  4),
])
